packages/lab/src/input-next/InputNext.tsx

```diff
@@ -0,0 +1,222 @@
+   */
+  disabled?: HTMLInputElement["disabled"];
+  /**
+   * The HTML element to render the Input, e.g. 'input', a custom React component.
+   */
+  inputComponent?: ElementType;
+  /**
+   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
+   */
+  inputProps?: InputHTMLAttributes<HTMLInputElement>;
+  onBlur?: FocusEventHandler<HTMLInputElement>;
+  /**
+   * Callbacks for events.
+   */
+  onChange?: (event: ChangeEvent<HTMLInputElement>, value: string) => void;
+  onFocus?: FocusEventHandler<HTMLInputElement>;
+  onKeyDown?: KeyboardEventHandler<HTMLInputElement>;
+  onKeyUp?: KeyboardEventHandler<HTMLInputElement>;
+  onMouseUp?: MouseEventHandler<HTMLInputElement>;
+  onMouseMove?: MouseEventHandler<HTMLInputElement>;
+  onMouseDown?: MouseEventHandler<HTMLInputElement>;
+  /**
+   * If `true`, the component is read only.
+   */
+  readOnly?: boolean;
+  type?: HTMLInputElement["type"];
+  /**
+   * The value of the `input` element, required for a controlled component.
+   */
+  value?: HTMLInputElement["value"];
+  /**
+   * Styling variant. Defaults to "primary".
+   */
+  variant?: "primary" | "secondary";
+}
+
+function mergeA11yProps(
+  a11yProps: Partial<
+    ReturnType<typeof useFormFieldPropsNext>["a11yProps"]
+  > = {},
+  inputProps: InputProps["inputProps"] = {},
+  misplacedAriaProps: AriaAttributes
+) {
+  const ariaLabelledBy = clsx(
+    a11yProps["aria-labelledby"],
+    inputProps["aria-labelledby"]
+  );
+
+  return {
+    ...misplacedAriaProps,
+    ...a11yProps,
+    ...inputProps,
+    // TODO: look at this - The weird filtering is due to TokenizedInputBase
+    "aria-labelledby": ariaLabelledBy
+      ? Array.from(new Set(ariaLabelledBy.split(" "))).join(" ")
+      : null,
+  };
+}
+
+export const Input = forwardRef<HTMLInputElement, InputProps>(function Input(
+  {
+    "aria-activedescendant": ariaActiveDescendant,
+    "aria-expanded": ariaExpanded,
+    "aria-owns": ariaOwns,
+    className: classNameProp,
+    disabled,
+    id,
+    inputComponent: InputComponent = "input",
+    inputProps: inputPropsProp,
+    role,
+    style,
+    value: valueProp,
+    // If we leave both value and defaultValue undefined, we will get a React warning on first edit
+    // (uncontrolled to controlled warning) from the React input
+    defaultValue: defaultValueProp = valueProp === undefined ? "" : undefined,
+    onBlur,
+    onChange,
+    onFocus,
+    onKeyDown,
+    onKeyUp,
+    onMouseUp,
+    onMouseMove,
+    onMouseDown,
+    readOnly: readOnlyProp,
+    type = "text",
+    variant = "primary",
+    ...other
+  },
+  ref
+) {
+  const {
+    a11yProps: {
+      disabled: a11yDisabled,
+      readOnly: a11yReadOnly,
+      ...restA11y
+    } = {},
+  } = useFormFieldPropsNext();
+
+  const isDisabled = disabled || a11yDisabled;
+  const isReadOnly = readOnlyProp || a11yReadOnly;
+
+  const [focused, setFocused] = useState(false);
```

Comment:
don't think we need `focused`

Reply:
this is used for the styling since tab focuses on the inner input rather than the outer div. the CSS is a bit funky if we want to do it the other way?

Reply:
this should work

```
-.saltInputNext-focused {
+.saltInputNext:has(:focus) {
```

Reply:
are we allowed to use has yet?